packages/codebytes/src/__tests__/codebyte-test.tsx

```diff
@@ -0,0 +1,197 @@
+import './mocks';
+
+import { setupRtl } from '@codecademy/gamut-tests';
+import userEvent from '@testing-library/user-event';
+import { encode } from 'js-base64';
+import React from 'react';
+
+import { CodeByteEditor } from '..';
+import { helloWorld, validLanguages } from '../consts';
+import { trackClick } from '../helpers';
+import { trackUserImpression } from '../libs/eventTracking';
+import { CodeByteEditorProps } from '../types';
+
+const mockEditorTestId = 'mock-editor-test-id';
+
+// This is a super simplified mock capable of render value and trigger onChange.
+jest.mock('../MonacoEditor', () => ({
+  SimpleMonacoEditor: ({
+    value,
+    onChange,
+  }: {
+    value: string;
+    onChange?: (value: string) => void;
+  }) => (
+    <>
+      {value}
+      <input
+        data-testid={mockEditorTestId}
+        type="text"
+        onChange={(e) => {
+          onChange?.(e.target.value);
+        }}
+        value={value}
+      />
+    </>
+  ),
+}));
+
+const renderWrapper = setupRtl(CodeByteEditor, {});
+
+type RenderWrapperWithProps = CodeByteEditorProps & { mode?: string };
+
+const renderWrapperWith = ({ mode, ...rest }: RenderWrapperWithProps) => {
+  const url = new URL(window.location.href);
+
+  const { text, language } = rest;
+  if (text) {
+    url.searchParams.set('text', encode(text));
+  }
+  if (language) {
+    url.searchParams.set('lang', language);
+  }
+  url.searchParams.set('client-name', 'forum');
+  url.searchParams.set(
+    'page',
+    'https://discuss.codecademy.com/some-interesting/post'
+  );
+  if (mode) {
+    url.searchParams.set('mode', mode);
+  }
+  window.history.replaceState({}, '', url.toString());
+
+  return renderWrapper(rest);
+};
```

Comment:
This function sets query params on the Codebytes component for testing purposes. By using 'rest' here, we are forwarding all the props from the Codebytes component, while also allowing mode to be set to the query params value.